pkg/cmd/azurekeyvaulttemplatefuncs.go

```diff
@@ -0,0 +1,47 @@
+package cmd
+
+import (
+	"context"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+	"github.com/Azure/azure-sdk-for-go/sdk/security/keyvault/azsecrets"
+)
+
+type azureKeyVaultConfig struct {
+	VaultURL string `json:"vaultUrl" mapstructure:"vaultUrl" yaml:"vaultUrl"`
+	client   *azsecrets.Client
+	cache    map[string]string
+}
+
+func (c *Config) azureKeyVaultTemplateFunc(name string) string {
+	if secret, ok := c.AzureKeyVault.cache[name]; ok {
```

Comment:
The cache key *should* be set based on the secret name and the vault URL, especially if the suggestion is accepted to allow multiple vault URLs.

Reply:
This is only needed if multiple vaults are permitted.